### PR TITLE
🗞️ Add mint and burn oft adapter example

### PR DESCRIPTION
**Pull request by St0rmBr3w**
Dependant on #1116

### CODE REVIEW

File: examples/mint-burn-oft-adapter/CHANGELOG.md
Comment (shankars99) [inline examples/mint-burn-oft-adapter/CHANGELOG.md:1]: is this a new example? if so we can remove changelog

Comment (St0rmBr3w) [reply examples/mint-burn-oft-adapter/CHANGELOG.md:1]: The changeset is from the oft-evm package changes, not the example.

File: package.json [hunk 1]
```diff
@@ -40,9 +40,9 @@
     "@changesets/cli": "^2.27.1",
     "@layerzerolabs/prettier-config-next": "^2.3.39",
     "@types/node": "~18.18.14",
-    "@typescript-eslint/eslint-plugin": "^7.5.0",
-    "@typescript-eslint/parser": "^7.5.0",
-    "eslint": "^8.55.0",
+    "@typescript-eslint/eslint-plugin": "^7.7.1",
```
[the remark this answers is not in the document]
Comment (St0rmBr3w) [reply package.json:43]: I could not commit for some reason without bumping this, linter was failing on all examples without bump.

File: examples/mint-burn-oft-adapter/contracts/MyOFT.sol [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.22;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { OFT } from "@layerzerolabs/oft-evm/contracts/OFT.sol";
+
+contract MyOFT is OFT {
```
[the remark this answers is not in the document]
Comment (St0rmBr3w) [reply examples/mint-burn-oft-adapter/contracts/MyOFT.sol:7]: I don't think this is true, this isn't a mock contract, but the actual MyOFT example.

For example, this is the oft example:

#https://github.com/LayerZero-Labs/devtools/blob/5c47a41148a40d894acc305deb4e6609fad06049/examples/oft/contracts/MyOFT.sol#L6-L7

File: examples/mint-burn-oft-adapter/contracts/MyMintBurnOFTAdapter.sol [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: UNLICENSED
```
[the remark this answers is not in the document]
Comment (St0rmBr3w) [reply examples/mint-burn-oft-adapter/contracts/MyMintBurnOFTAdapter.sol:1]: Not for this contract, in `./packages/oft-evm/MintBurnOFTAdapter.sol` we should.

**Review by shankars99 — APPROVED** [review]

**Pull request merged**